docs/source/transforms.rst

```diff
@@ -5,242 +5,449 @@ Transforming and augmenting images
 
 .. currentmodule:: torchvision.transforms
 
+Torchvision supports common computer vision transformations in the
```

Comment:
All the text below is mostly new. There was very little copy/pasting. Mostly re-writing from scratch.